privacyidea/static/components/token/controllers/tokenControllers.js

```diff
@@ -294,6 +296,11 @@ myApp.controller("tokenEnrollController", ["$scope", "TokenFactory", "$timeout",
         } else {
             $scope.form.genkey = true;
         }
+        if ($scope.form.type === "yubikey") {
+            delete $scope.form.otplen;
+        } else {
+            $scope.form.otplen = 6;
```

Comment:
Are there side effects like?

Currently if I would change from HOTP with otplen=8 to TOTP, I would keep the otplen=8. Very convinient!

In this case, we would **always reset** to 6. This is a change in behaviour and inconvenient for 99% of the users, who are not using tokentype yubikey anyways. Is there a way to avoid this?

Reply:
The general `else`-part makes not sense, correct. But we should think about what we want to achieve here:
If the user switches from `HOTP` with `otplen=8` to `Yubikey AES` (which only supports `otplen=[32, 44, 48]`) the selected `otplen`-dropdown would be empty (and no `otplen` parameter would send in the `/token/init` request).
If the user then switches back to `TOTP` after changing the `otplen` to e.g. `44`, the `otplen`-dropdown would be empty again.

Reply:
Fixed in 46885a7